packages/cli/src/commands/env/pull.ts

```diff
@@ -65,6 +66,15 @@ export default async function pull(
     return 1;
   }
 
+  const gitBranch = opts['--branch'];
+
+  if (gitBranch && environment !== 'preview') {
+    output.error(
+      `The \`--branch\` flag can only be used with \`--environment preview\``
```

Comment:
I think we should throw this error in the API instead right?

Reply:
API now handles this:

```
 $ vc pull --branch old --environment production
Vercel CLI 28.7.2
> Overwriting existing .vercel/.env.production.local file
> Downloading `production` Environment Variables for Project n8.io
Error: Invalid request: `target` must be "preview" when specifying a `gitBranch`
```

Reply:
@TooTallNate Lets remove this `if` statement then?